types/user.ts

```diff
@@ -251,6 +259,130 @@ export class User extends LazyObject {
     console.log('Auth deleted');
   }
 
+  /** clearAllReports
+   * Clear all reports on this piece of content.
+   * @param content: the post, comment, or user to be cleared of all current reports
+   */
+  public async clearAllReports(content: Post | Comment | User) {
+    await this.checkIfSignedIn();
+    if (this.status! < UserStatus.Employee) return Promise.reject('Not an employee, cant moderate!');
+
+    if (!content.hasData) await content.getData();
+
+    // update client side by deleting all reports from the content's report array
+    content.reports = content.reports?.filter(
+      (report) => !refEqual(report.docRef!, this.docRef!)
+    );
+
+    // update server side: get all reports on this content from the reports collection
+    const q = await getDocs(
+      query(
+      collection(db, 'reports'),
+      where('content', '==', content.docRef!)
+      )
+    );
+    // run a transaction that deletes all reports found in this query
+    return runTransaction(db, async (transaction: Transaction) => {
```

Comment:
this is probably fine... it'll technically fail if there's more than 250 reports on a content, but like, will we even have that many users? lol

Reply:
We could also just have a thing where if a content has been reported more than 10 times, then just don't report it again. That actually seems better for employees, cuz tbh if something's been automoderated, why bother with making more reports? I'll file an issue for that